lib/spoom/deadcode/remover.rb

```diff
@@ -0,0 +1,616 @@
+            #  :baz,
+            # )
+            # ~~~
+            delete_lines(context.node.location.start_line, context.node.location.end_line)
+          elsif prev_node && next_node
+            # We have a node before and after one the same line, just remove the part of the line
+            #
+            # ~~~
+            # attr_reader :foo, :bar, :baz # we need to remove bar
+            # ~~~
+            replace_chars(prev_node.location.end_char, next_node.location.start_char, ", ")
+          elsif prev_node
+            # We have a node before, on the same line, but no node after, just remove the part of the line
+            #
+            # ~~~
+            # attr_reader :foo, :bar, :baz # we need to remove baz
+            # ~~~
+            delete_chars(prev_node.location.end_char, context.node.location.end_char)
+          elsif next_node
+            # We don't have a node before but a node after on the same line, just remove the part of the line
+            #
+            # ~~~
+            # attr_reader :foo, :bar, :baz # we need to remove foo
+            # ~~~
+            delete_chars(context.node.location.start_char, next_node.location.start_char)
+          else
+            raise "Unexpected case while removing attr_accessor"
+          end
+
+          insert_accessor(context.node, send_context, was_removed: false) if need_accessor
+        end
+
+        sig do
+          params(
+            node: SyntaxTree::Node,
+            send_context: NodeContext,
+            was_removed: T::Boolean,
+          ).void
+        end
+        def insert_accessor(node, send_context, was_removed:)
+          name = @node_context.node_string(node)
+          code = case @kind
+          when Definition::Kind::AttrReader
+            "attr_writer #{name}"
+          when Definition::Kind::AttrWriter
+            "attr_reader #{name}"
+          end
+
+          indent = " " * send_context.node.location.start_column
+
+          sig = send_context.attached_sig
+          sig_string = transform_sig(sig, name: name, kind: @kind) if sig
+
+          node_after = send_context.next_node
+
+          if was_removed
+            first_node = send_context.attached_comments_and_sigs.first || send_context.node
+            at_line = first_node.location.start_line - 1
+
+            prev_context = NodeContext.new(@old_source, first_node, send_context.nesting)
+            node_before = prev_context.previous_node
+
+            new_line_before = node_before && send_context.node.location.start_line - node_before.location.end_line < 2
+            new_line_after = node_after && node_after.location.start_line - send_context.node.location.end_line <= 2
+          else
+            at_line = send_context.node.location.end_line
+            new_line_before = true
+            new_line_after = node_after && node_after.location.start_line - send_context.node.location.end_line < 2
+          end
+
+          lines_to_insert = []
```

Comment:
NIT: since we're not using any characters to join, we could just do a string here
```suggestion
          lines_to_insert = +""
```